CorsixTH/Lua/app.lua

```diff
@@ -1750,6 +1739,20 @@ function App:isAudioEnabled()
   return TH.GetCompileOptions().audio
 end
 
+function App:getSystemInfo()
+  local compile_opts = TH.GetCompileOptions()
+  local comp_details = ""
+  for key, value in pairs(compile_opts) do
```

Comment:
If we want a consistent order we'd need to build a list of keys and sort it (or hardcode it, but that's not as nice is it?)

Reply:
Python2 also had non-deterministic dictionaries, and settled in Python3 for deterministic insertion order. Not sorted, but faster.

Simplest long term solution would be to change the lua table itself, but that may be out of scope, as you don't have a standard lua implementation any more in that case.

Sorting will work for tables with sortable keys, but likely not everything is fully ordered.

Reply:
I'm talking about a particular case where the UI (or gamelog header in this case) depended on the order of the pairs. That's a broader discussion - factorio for one modified lua's table to have a deterministic order, but that's likely out of scope for us. We can hack it with lists of keys where we need to (perhaps encapsulated)